src/scripts/summary.js

```diff
@@ -410,17 +410,29 @@ class Summary extends H5P.EventDispatcher {
     for (const section of sections) {
       const sectionRow = document.createElement("li");
       sectionRow.classList.add('h5p-interactive-book-summary-overview-section-details');
+      const sectionRowContainer = document.createElement("div");
+      sectionRowContainer.classList.add('h5p-interactive-book-summary-overview-section-container');
+      sectionRow.appendChild(sectionRowContainer);
       if (this.behaviour.progressIndicators) {
         const icon = document.createElement("span");
         icon.classList.add('h5p-interactive-book-summary-section-icon');
         icon.classList.add(section.taskDone ? 'icon-chapter-done' : 'icon-chapter-blank');
-        sectionRow.appendChild(icon);
+        sectionRowContainer.appendChild(icon);
       }
 
       const title = document.createElement("button");
       title.type = "button";
       title.classList.add('h5p-interactive-book-summary-section-title');
       title.onclick = () => {
+        if (section.instance.libraryInfo.machineName === 'H5P.ActiveReaderTextInput') {
+          const contentChapter = document.querySelector(`.h5p-content-chapter-${chapterId}`);
+          // toggle hidden class on the chapter
+          contentChapter.classList.toggle('hidden');
+
+          // Resize the iframe
+          setTimeout(() => this.parent.trigger('resize'), 200);
```

Comment:
Function bodies that are not just returning a value, even if it's one line only, should be written in braces:

```
setTimeout(() => {
  this.parent.trigger('resize');
}, 200);
```

What's the timeout needed for?

Reply:
The reason for this is the slider that opens up on click has some transition effect on height so we need to wait when the div expanded fully and then initiate the parent resize.

Reply:
Then there are three better solutions at least:
a) At least add a comment what the magic number is.
b) Replace the magic number by a constant and add a comment to the constant.
c) Add a `transtitionend` listener to the `responseField` and let that one trigger the resize. Not only will you get rid of the magic number (and thus not require to change it whenever the transition time is changed) and the timeout, but also ensure that any trigger for the `responseField` to toggle its transition will trigger a resize not just a click - may never be needed here, but would be the general cover-all solution. Along with that, you may want to check the `propertyName` of the `transitionend` event though, because you used `transition all` which is a little wonky in itself to let all CSS properties transition and not just one that controls the height (there's probably some other better "accordion" solution).

Reply:
a) I will add a comment
b) I do not think we need to define the constant for a single operation if that was repetitive I would have created an event. A comment will be sufficient enough to explain the necessity of the code.
c) I do not it is a good idea to add a transition event listener for each field we use in content type, imagine the user has 4 inputs then we are ended up adding 4 event listeners. We are just increasing application overhead by adding more event listeners. In addition, the outcome would be the same. Hence,  I would like to stick with the magic number :-)
I use transition all CSS because there is height, margin, padding and border we need to add a transition for all of them, rather than that I use transition all which seems a good option (in future if we change the design then the developer does not have to focus on the transition that's a plus)

Just a thought, if possible, please try to give all feedback at once so that we can avoid many iterations :-)
Or you can contact me directly on Slack so that we can be more effective to complete this task.

Reply:
Let's keep this one at least on github. It's a potential source for learning that others can use, and that's what NDLA wants. It's not common practice, but there are people who argue that not only the product should be open and transparent, but the process itself. It helps to let others understand HOW creating code works in practice instead of just seeing the result. Yes, that can mean a little more communication. @janlindso can gladly overrule me here by the way :-)

You do not need to add a `transitionend` listener/handler for every field if you're worried about performance. That's what the JavaScript gods gave us event delegation for. Events will bubble up unless propagation is stopped. You can add one single `transitionend` listener to a parent element as well, e.g. `h5p-interactive-book-summary-overview-list` and handle it there. The "outcome" will be the same, but future maintenance will be decreased. If someone wants to change the transition time later on, he/she will not have to know that this needs to be done in two places (CSS and JavaScript).

Having constants is not only useful to not repeat oneself, but to keep the code clean and to make maintenance easier. If all constants are defined in the same place, e.g. the top or bottom of the code, it's easier to find them - and you even have the benefit of naming them in order to reveal their purpose, e.g. `const responseFieldtransitionTimeMS = 200;`.

Reply:
Thank you for explaining. No need to overrule your decision, I understood the suggested approach is more sensible for users. I have updated the code, please have a look.

Reply:
That should do. Good job! :+1:

If my picky mind finds something later on (the original code of Interactive Book is a little wacky anyway), I can tweak it myself ...